app/scripts/components/common/catalog/utils.ts

```diff
@@ -0,0 +1,54 @@
+import { omit, set } from 'lodash';
+
+export enum CatalogActions {
+  CLEAR = 'clear',
+  SEARCH = 'search',
+  TAXONOMY_MULTISELECT = 'taxonomy_multiselect'
+}
+
+export type CatalogViewAction = (action: CatalogActions, value?: any) => void;
+
+export function onCatalogAction(
+  action: CatalogActions,
+  value: any,
+  taxonomies: any,
+  setSearch: (value: string) => void,
```

Comment:
Oh! I realize that we would need more of this type of utility function - which takes either `setState` or `setAtom`as a setter function - like the one that we use for reconciling STAC metadata here: https://github.com/NASA-IMPACT/veda-ui/pull/988/files#diff-b0d6bd8ff29016fd81054f35bfcc97dd5b50077310a81f6043a99240cb43a4eaR167

I'd like to ask your thoughts how we should type these util functions moving forward. Should we explicitly type the setter function as either setAtom or setState? ex. I tried this method in  https://github.com/NASA-IMPACT/veda-ui/pull/988/files#diff-b0d6bd8ff29016fd81054f35bfcc97dd5b50077310a81f6043a99240cb43a4eaR169 - and I am still unsure if this just added complicacy or not. But I also think `any` is a bit too broad. Any thoughts?

Reply:
I think both can be verbose, even for simpler types like a setter for a string (ie. `React.Dispatch<React.SetStateAction<string>>`).

How about we create aliases for the setters and use that? For example: `type SetState<T> = Dispatch<SetStateAction<T>>` and then we can use it like this:

`const [count, setCount]: [number, SetState<number>] = useState(0);`

Reply:
Yeah, aliases sound great. I will leave it up to you if you want to apply this change to this pr or not.